beacon-chain/db/filesystem/blob.go

```diff
@@ -279,10 +279,22 @@ func (bs *BlobStorage) Prune(pruneBefore primitives.Slot) error {
 // processFolder will delete the folder of blobs if the blob slot is outside the
 // retention period. We determine the slot by looking at the first blob in the folder.
 func (bs *BlobStorage) processFolder(folder string, pruneBefore primitives.Slot) (int, error) {
-	f, err := bs.fs.Open(filepath.Join(folder, "0."+sszExt))
-	if err != nil {
-		return 0, err
+	var f afero.File
+	var err error
+	var found bool
+
+	for i := 0; i < fieldparams.MaxBlobsPerBlock; i++ {
+		f, err = bs.fs.Open(filepath.Join(folder, fmt.Sprintf("%d.%s", i, sszExt)))
```

Comment:
This creates dangling file descriptors (only the last one is closed).

Consider instead calling `slotFromBlob` in the loop (requires you to predeclare `slot` so it has outer scope) and closing `f` at end of each iteration.

Reply:
Done... instead of `found`, I switched to checking `slot == 0`, let me know if want me to change that

Reply:
Edge case would be e2e/devnet if starting with a genesis block that has blobs - although I'm guessing we never start from such a block, since that would add more dependencies to node startup, so it's probably a non-issue.

Reply:
Also had to fix a bad unit test assumption